extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -118,13 +120,26 @@ public PolarisEclipseLinkMetaStoreSessionImpl(
   }
 
   /** Load the persistence unit properties from a given configuration file */
-  private Map<String, String> loadProperties(String confFile, String persistenceUnitName) {
+  private Map<String, String> loadProperties(
+      @NotNull String confFile, @NotNull String persistenceUnitName) {
     if (this.properties != null) {
       return this.properties;
     }
 
     try {
-      InputStream input = this.getClass().getClassLoader().getResourceAsStream(confFile);
+      InputStream input = null;
+      int splitPosition = confFile.indexOf("!/");
+      if (splitPosition == -1) {
+        input = this.getClass().getClassLoader().getResourceAsStream(confFile);
+      } else {
+        String jarPrefixPath = confFile.substring(0, splitPosition);
+        String descPath = confFile.substring(splitPosition + 2);
+        URL prefixUrl = this.getClass().getClassLoader().getResource(jarPrefixPath);
+        URLClassLoader child =
+            new URLClassLoader(new URL[] {prefixUrl}, this.getClass().getClassLoader());
+        input = child.getResourceAsStream(descPath);
```

Comment:
I guess I don't understand what's going on. This `InputStream` is _not_ being used to setup eclipselink? Cause this `InputStream` could be coming from anywhere. Why is there a restriction that the resource is on the classpath?

Reply:
This _is_ used to parse it, but then EclipseLink again tries to load the resource directly when we call `Persistence.createEntityManagerFactory(persistenceUnitName, properties)`

Reply:
This _is_ used to parse it, but then EclispeLink again tries to load the resource directly when we call `Persistence.createEntityManagerFactory`

Reply:
Since each realm has its only database, I have the following configuration in persistence.xml. I read this configuration and replace the realm then pass it to `Persistence.createEntityManagerFactory()` which will again read persistence.xml for the rest of the configurations.  `Persistence.createEntityManagerFactory()` assumes the persistence.xml is a resource file inside a jar.     

<property name="jakarta.persistence.jdbc.url"  value="jdbc:h2:file:/tmp/eclipseLink/{realm}"/>